objects/src/accounts/account_id.rs

```diff
@@ -0,0 +1,200 @@
+    pub const FUNGIBLE_FAUCET_TAG: u32 = 0;
+    pub const NON_FUNGIBLE_FAUCET_TAG: u32 = 1 << 31;
+
+    pub const MIN_TRAILING_ZEROS: u32 = 24;
+
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+    /// Returns a new account ID derived from the specified seed.
+    ///
+    /// The account ID is computed by hashing the seed and using 3 elements of the result to form
+    /// the ID. Specifically we are take elements 0, 1, and 3, omitting element 2. We omit element
+    /// 2 because unlike elements 0 and 3, it has no special structure which we need to carry over
+    /// into the derived account ID. Element 1 could have been omitted just as well.
+    ///
+    /// # Errors
+    /// Returns an error if the resulting account ID does not comply with account ID rules.
+    pub fn new(seed: Word) -> Result<Self, AccountError> {
+        // hash the seed and build the ID from the 1st, 2nd, and 4th elements of the result
+        let hash = Hasher::hash_elements(&seed);
+        let id = Self(hash[0], hash[1], hash[3]);
+
+        // verify that the ID satisfies all rules
+        id.validate()?;
+
+        Ok(id)
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns true if an account with this ID can issue fungible assets.
+    pub fn is_fungible_faucet(&self) -> bool {
+        self.tag() == Self::FUNGIBLE_FAUCET_TAG
+    }
+
+    /// Returns true if an account with this ID can issue non-fungible assets.
+    pub fn is_non_fungible_faucet(&self) -> bool {
+        self.tag() == Self::NON_FUNGIBLE_FAUCET_TAG
+    }
+
+    /// Returns true if an account with this ID can issue assets.
+    pub fn is_faucet(&self) -> bool {
+        self.is_fungible_faucet() || self.is_non_fungible_faucet()
+    }
+
+    // SEED GENERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Finds and returns a seed suitable for creating regular account IDs using the provided seed
+    /// as a starting point.
+    pub fn get_account_seed(_init_seed: [u8; 32]) -> Word {
+        todo!()
+    }
+
+    /// Finds and returns a seed suitable for creating account IDs for fungible faucets using the
+    /// provided seed as a starting point.
+    pub fn get_fungible_faucet_seed(_init_seed: [u8; 32]) -> Word {
+        todo!()
+    }
+
+    /// Finds and returns a seed suitable for creating account IDs for non-fungible faucets using
+    /// the provided seed as a starting point.
+    pub fn get_non_fungible_faucet_seed(_init_seed: [u8; 32]) -> Word {
+        todo!()
+    }
+
+    // HELPER METHODS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the first bit of this account ID.
+    fn first_bit(&self) -> u8 {
```

Comment:
I think we are almost always working with canonical representations - so, I think it is fine to omit it here. But in cases when we work with non-canonical representations it would probably be a good idea to add prefixes and comments.